Tests/BSONTests/BSONTests.swift

```diff
@@ -71,26 +71,42 @@ func toByteString(_ bytes: [UInt8]?) -> String {
 }
 
 public extension Data {
-    var byteString: String {
-        BSONTests.toByteString([UInt8](self))
+    func toByteString(ascii: Bool = true) -> String {
+        BSONTests.toByteString([UInt8](self), ascii: ascii)
     }
 }
 
 public extension Array where Element == UInt8 {
-    var byteString: String {
-        BSONTests.toByteString(self)
+    func toByteString(ascii: Bool = true) -> String {
+        BSONTests.toByteString(self, ascii: ascii)
+    }
+
+    init?(hexString: String) {
```

Comment:
this looks very similar to our logic for creating an ObjectID from a hex string. Perhaps we need some internal helper method that we can use in both places?